src/coerce.c

```diff
@@ -6,11 +6,31 @@
 #include "conditions.h"
 
 void cant_coerce(SEXP from, SEXP to, int i) {
+
+  const char* to_friendly;
+  switch(TYPEOF(to)) {
+    case INTSXP:
+      to_friendly = "an integer";
+      break;
+    case REALSXP:
+      to_friendly = "a double";
+      break;
+    case STRSXP:
+      to_friendly = "a string";
+      break;
+    case LGLSXP:
+      to_friendly = "a logical";
+      break;
```

Comment:
Just to keep things in sync, it looks like `set_vector_value()` does call `cant_coerce()` for `RAWSXP` too (but not for `VECSXP`, so no need to add that case)